openslides_backend/action/actions/speaker/create_update_delete.py

```diff
@@ -199,6 +199,65 @@ class SpeakerUpdate(UpdateAction):
     schema = DefaultSchema(Speaker()).get_update_schema(["speech_state"])
     permission = Permissions.ListOfSpeakers.CAN_MANAGE
 
+    def update_instance(self, instance: Dict[str, Any]) -> Dict[str, Any]:
+        speaker = self.datastore.get(
+            FullQualifiedId(self.model.collection, instance["id"]),
+            ["speech_state", "meeting_id"],
+        )
+        meeting = self.datastore.get(
+            FullQualifiedId(Collection("meeting"), speaker["meeting_id"]),
+            [
+                "list_of_speakers_can_set_contribution_self",
+                "list_of_speakers_enable_pro_contra_speech",
+            ],
+        )
+        has_can_manage = has_perm(
+            self.datastore,
+            self.user_id,
+            Permissions.ListOfSpeakers.CAN_MANAGE,
+            speaker["meeting_id"],
+        )
+        allowed_self_contribution = has_can_manage or meeting.get(
+            "list_of_speakers_can_set_contribution_self"
+        )
+        allowed_pro_contra = meeting.get("list_of_speakers_enable_pro_contra_speech")
+        msg = "Not allowed to set speech_state"
+        if speaker.get("speech_state") == instance.get("speech_state"):
+            pass
+        elif instance.get("speech_state") == "contribution":
+            if not allowed_self_contribution:
+                raise ActionException(msg)
```

Comment:
Maybe a fitting message describing the situation would be better. E.g. here: "Self contributing is not enabled."

Reply:
Done.